certification/internal/shell/base_on_ubi.go

```diff
@@ -32,6 +29,10 @@ func (p *BaseOnUBICheck) validate(podmanEngine cli.PodmanEngine, image string, l
 
 	lines := strings.Split(runReport.Stdout, "\n")
 
+	return p.checkBasedOnUbi(lines)
+}
+
+func (p *BaseOnUBICheck) checkBasedOnUbi(lines []string) (bool, error) {
```

Comment:
Can we refactor this into a generic function (sth like `parse`) and have it included in the interface? It seems like all of our tests need to parse a command's output.

Reply:
Not all do that. And I'd like to see the output from the engine become more structured. See #2

Reply:
+1 to @samira-barouti's request here. If we have multiple validations that require "parsing the output of `containercli run ...` then we probably would see a benefit of abstracting that out and testing that independently. 

Could we have that drafted into a new issue+PR?

---
Separately, bit of a nitpick regarding naming - I'd prefer to keep this as `validate` over `checkBasedOnUbi`. I don't see an issue with reusing the public method name as a private method name with the internal working defined.

Reply:
If it ends up being part of the interface, that's great. But I don't think we necessarily need to add it just yet. Let's see how many end up actually needing it.